File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/PubliseringsdatoerImportService.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon;

import lombok.extern.slf4j.Slf4j;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.datavarehus.DatavarehusRepository;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domene.PubliseringsdatoDbDto;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.database.PubliseringsdatoerRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Slf4j
@Component
public class PubliseringsdatoerImportService {

  private final PubliseringsdatoerRepository publiseringsdatoerRepository;
  private final DatavarehusRepository datavarehusRepository;

  public PubliseringsdatoerImportService(
      PubliseringsdatoerRepository publiseringsdatoerRepository,
      DatavarehusRepository datavarehusRepository) {
    this.publiseringsdatoerRepository = publiseringsdatoerRepository;
    this.datavarehusRepository = datavarehusRepository;
  }

  public void importerDatoerFraDatavarehus() {
    List<PubliseringsdatoDbDto> publiseringsdatoerFraDvh =
        datavarehusRepository.hentPubliseringsdatoerFraDvh();

    publiseringsdatoerRepository.oppdaterPubliseringsdatoer(publiseringsdatoerFraDvh);
  }
}

File: src/main/kotlin/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/publiseringsdatoApi/PubliseringsdatoerImportService.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.publiseringsdatoApi

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.database.PubliseringsdatoerRepository
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.datavarehus.DatavarehusRepository
import org.springframework.stereotype.Component

@Component
class PubliseringsdatoerImportService(
    private val publiseringsdatoerRepository: PubliseringsdatoerRepository,
    private val datavarehusRepository: DatavarehusRepository
) {
    fun importerDatoerFraDatavarehus() {
        val publiseringsdatoerFraDvh = datavarehusRepository.hentPubliseringsdatoerFraDvh()
        publiseringsdatoerRepository.oppdaterPubliseringsdatoer(publiseringsdatoerFraDvh)
    }
}
